<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from TypeA.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="awesome-o" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.11 0.05 0.12"/>
      </geometry>
    </visual>
  </link>
  <link name="R_shoulder">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
  </link>
  <joint name="R_SAA" type="revolute">
    <parent link="base_link"/>
    <child link="R_shoulder"/>
    <origin rpy="1.57079632679 0 0" xyz="0.07 0 0.03"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-3.1415926536" upper="3.1354567304" velocity="1.0"/>
  </joint>
  <link name="R_upper_arm">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="green">
        <color rgba="0 0.8 0 1"/>
      </material>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="R_SFE" type="revolute">
    <parent link="R_shoulder"/>
    <child link="R_upper_arm"/>
    <origin rpy="0 -1.5 0" xyz="0.00 0 0.0"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-2.0371264863" upper="1.8898643307" velocity="1.0"/>
  </joint>
  <link name="R_forearm">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="R_EB" type="revolute">
    <parent link="R_upper_arm"/>
    <child link="R_forearm"/>
    <origin rpy="0 0 0" xyz="0.00 0 -0.08"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-2.1291653336" upper="2.1353012567" velocity="1.0"/>
  </joint>
  <link name="L_shoulder">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
  </link>
  <joint name="L_SAA" type="revolute">
    <parent link="base_link"/>
    <child link="L_shoulder"/>
    <origin rpy="1.57079632679 0 0" xyz="-0.07 0 0.03"/>
    <axis xyz="-1 0 0"/>
    <limit effort="1000.0" lower="-3.1415926536" upper="3.1354567304" velocity="1.0"/>
  </joint>
  <link name="L_upper_arm">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="green">
        <color rgba="0 0.8 0 1"/>
      </material>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="L_SFE" type="revolute">
    <parent link="L_shoulder"/>
    <child link="L_upper_arm"/>
    <origin rpy="0 1.5 0" xyz="0.00 0 0.0"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-1.9144080233" upper="1.9512235622" velocity="1.0"/>
  </joint>
  <link name="L_forearm">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="L_EB" type="revolute">
    <parent link="L_upper_arm"/>
    <child link="L_forearm"/>
    <origin rpy="0 0 0" xyz="0.00 0 -0.08"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-2.0984857178" upper="2.0739420252" velocity="1.0"/>
  </joint>
  <link name="R_hip">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
  </link>
  <joint name="R_HAA" type="revolute">
    <parent link="base_link"/>
    <child link="R_hip"/>
    <origin rpy="0 0 0" xyz="0.03 0 -0.06"/>
    <axis xyz="0 0 -1"/>
    <limit effort="1000.0" lower="-2.1966604883" upper="0.5645049299" velocity="1.0"/>
  </joint>
  <link name="R_hip_split">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="R_HR" type="revolute">
    <parent link="R_hip"/>
    <child link="R_hip_split"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-1.4848934027" upper="0.9694758579" velocity="1.0"/>
  </joint>
  <link name="R_upper_leg">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="green"/>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="R_HFE" type="revolute">
    <parent link="R_hip_split"/>
    <child link="R_upper_leg"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="-1.0 0 0"/>
    <limit effort="1000.0" lower="-2.8041168803" upper="0.6320000846" velocity="1.0"/>
  </joint>
  <link name="R_calf">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="R_KFE" type="revolute">
    <parent link="R_upper_leg"/>
    <child link="R_calf"/>
    <origin rpy="0 0 0" xyz="0 0 -0.08"/>
    <axis xyz="1.0 0 0"/>
    <limit effort="1000.0" lower="-2.5280003384" upper="0.0490873852" velocity="1.0"/>
  </joint>
  <link name="R_foot">
    <visual>
      <geometry>
        <box size="0.05 0.075 0.05"/>
      </geometry>
      <material name="green"/>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="R_AFE" type="revolute">
    <parent link="R_calf"/>
    <child link="R_foot"/>
    <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-0.5031456984" upper="2.0739420252" velocity="1.0"/>
  </joint>
  <link name="R_foot_pad">
    <visual>
      <geometry>
        <box size="0.06 0.10 0.005"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="R_AR" type="revolute">
    <parent link="R_foot"/>
    <child link="R_foot_pad"/>
    <origin rpy="0 0 0" xyz="0 0 -0.03"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-1.0246991663" upper="1.3989904786" velocity="1.0"/>
  </joint>
  <link name="L_hip">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
  </link>
  <joint name="L_HAA" type="revolute">
    <parent link="base_link"/>
    <child link="L_hip"/>
    <origin rpy="0 0 0" xyz="-0.03 0 -0.06"/>
    <axis xyz="0 0 -1"/>
    <limit effort="1000.0" lower="-1.2640001692" upper="2.1905245651" velocity="1.0"/>
  </joint>
  <link name="L_hip_split">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="L_HR" type="revolute">
    <parent link="L_hip"/>
    <child link="L_hip_split"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-1.0246991663" upper="1.4296700943" velocity="1.0"/>
  </joint>
  <link name="L_upper_leg">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="green"/>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="L_HFE" type="revolute">
    <parent link="L_hip_split"/>
    <child link="L_upper_leg"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="1.0 0 0"/>
    <limit effort="1000.0" lower="-0.5645049299" upper="2.6875343404" velocity="1.0"/>
  </joint>
  <link name="L_calf">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="L_KFE" type="revolute">
    <parent link="L_upper_leg"/>
    <child link="L_calf"/>
    <origin rpy="0 0 0" xyz="0 0 -0.08"/>
    <axis xyz="-1.0 0 0"/>
    <limit effort="1000.0" lower="-0.0736310778" upper="2.0739420252" velocity="1.0"/>
  </joint>
  <link name="L_foot">
    <visual>
      <geometry>
        <box size="0.05 0.075 0.05"/>
      </geometry>
      <material name="green"/>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="L_AFE" type="revolute">
    <parent link="L_calf"/>
    <child link="L_foot"/>
    <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    <axis xyz="-1 0 0"/>
    <limit effort="1000.0" lower="-2.0862138715" upper="0.5215534679" velocity="1.0"/>
  </joint>
  <link name="L_foot_pad">
    <visual>
      <geometry>
        <box size="0.06 0.10 0.005"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0 0 -0.045"/>
    </visual>
  </link>
  <joint name="L_AR" type="revolute">
    <parent link="L_foot"/>
    <child link="L_foot_pad"/>
    <origin rpy="0 0 0" xyz="0 0 -0.03"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-1.1167380136" upper="1.1228739367" velocity="1.0"/>
  </joint>
</robot>


